src/amo/components/RatingManager/index.js

```diff
@@ -82,53 +64,23 @@ type InternalProps = {|
 
 export class RatingManagerBase extends React.Component<InternalProps> {
   componentDidMount() {
-    const {
-      addon,
-      apiState,
-      loadSavedReview,
-      userId,
-      userReview,
-      version,
-    } = this.props;
+    const { addon, dispatch, errorHandler, userId, userReview } = this.props;
 
     if (userId && userReview === undefined) {
       log.debug(`Loading a saved rating (if it exists) for user ${userId}`);
-      loadSavedReview({
-        apiState,
-        userId,
-        addonId: addon.id,
-        addonSlug: addon.slug,
-        versionId: version.id,
-      });
+      dispatch(
+        fetchLatestUserReview({
+          userId,
+          addonId: addon.id,
+          addonSlug: addon.slug,
+          errorHandlerId: errorHandler.id,
+        }),
+      );
     }
   }
 
   onSelectRating = (score: number) => {
-    const {
-      addon,
-      apiState,
-      dispatch,
-      errorHandler,
-      userReview,
-      version,
-    } = this.props;
-
-    const params = {
-      errorHandler,
-      score,
-      apiState,
-      addonId: addon.id,
-      reviewId: undefined,
-      versionId: version.id,
-    };
-
-    if (userReview) {
-      log.debug(`Editing reviewId ${userReview.id}`);
-      params.reviewId = userReview.id;
-    } else {
-      log.debug(oneLine`Submitting a new review for
-        versionId ${params.versionId || '[empty]'}`);
-    }
+    const { addon, dispatch, errorHandler, userReview, version } = this.props;
```

Comment:
It doesn't show up in this diff, but I think you can also remove `version` from the `render` method of this component.

Reply:
The `render` function calls `invariant` to say that `version` is required. I think that's a good place to do so because you'll see it early on in development. FYI, `Addon` doesn't have Flow coverage and it still uses `RatingManager` so this invariant call is important.

Reply:
Why not just put the `invariant` in `onSelectRating`, where `version` is actually needed? I understand that having it in `render` is like an early warning system, but it still feels odd to me. At some point we won't even need `version` at all (when the back end is changed to just not care about versions when it comes to reviews). I guess it's fine to keep it in `render`, but I don't love it.

Reply:
I considered that but putting `invariant` in `onSelectRating` means that the error will only happen when a user tries to create a rating. That would be a crappy error to experience in real life. I know we do this in other places but that's only for `nullable` properties where it wouldn't make sense to throw in `render`. But, yeah, once we can change the backend for creating new ratings then the `version` requirement will go away.

Reply:
> once we can change the backend for creating new ratings...

I'm actually not sure we will ever be able to do that. I think it's important to know which version a user was referring to when they first rated an add-on.